packages/protocol/test/L1/TaikoL1.integration.test.ts

```diff
@@ -0,0 +1,160 @@
+import { expect } from "chai";
+import { BigNumber, ethers as ethersLib } from "ethers";
+import { ethers } from "hardhat";
+import { TaikoL1, TaikoL2 } from "../../typechain";
+import { BlockMetadata } from "../utils/block_metadata";
+import { commitBlock, generateCommitHash } from "../utils/commit";
+import { buildProposeBlockInputs, proposeBlock } from "../utils/propose";
+import { deployTaikoL1 } from "../utils/taikoL1";
+import { deployTaikoL2 } from "../utils/taikoL2";
+
+describe("integration:TaikoL1", function () {
+    let taikoL1: TaikoL1;
+    let taikoL2: TaikoL2;
+    let l2Provider: ethersLib.providers.JsonRpcProvider;
+    let l2Signer: ethersLib.Signer;
+
+    beforeEach(async function () {
+        l2Provider = new ethers.providers.JsonRpcProvider(
+            "http://localhost:28545"
+        );
+
+        l2Signer = await l2Provider.getSigner(
+            (
+                await l2Provider.listAccounts()
+            )[0]
+        );
+
+        taikoL2 = await deployTaikoL2(l2Signer);
+
+        const genesisHash = taikoL2.deployTransaction.blockHash as string;
+
+        taikoL1 = await deployTaikoL1(genesisHash);
+    });
+
+    describe("isCommitValid()", async function () {
+        it("should not be valid", async function () {
+            const block = await l2Provider.getBlock("latest");
+            const commit = generateCommitHash(block);
+
+            const isCommitValid = await taikoL1.isCommitValid(
+                1,
+                1,
+                commit.hash
+            );
+
+            expect(isCommitValid).to.be.eq(false);
+        });
+    });
+
+    describe("getProposedBlock()", function () {
+        it("proposed block does not exist", async function () {
```

Comment:
it("should return zero values for non-existing blocks without exception to be thrown")

Reply:
I would label this as a unit test, but maybe the difference is not important at all, that's why I think we should use the same setup for unit tests and integration tests.